baseplate/lib/message_queue.py

```diff
@@ -103,22 +134,39 @@ def put(self, message: bytes, timeout: Optional[float] = None) -> None:
         raise TimedOutError
 
     def unlink(self) -> None:
-        """Remove the queue from the system.
-
-        The queue will not leave until the last active user closes it.
-
-        """
         self.queue.unlink()
 
     def close(self) -> None:
-        """Close the queue, freeing related resources.
+        self.queue.close()
 
-        This must be called explicitly if queues are created/destroyed on the
-        fly. It is not automatically called when the object is reclaimed by
-        Python.
 
-        """
-        self.queue.close()
+class InMemoryMessageQueue(MessageQueue):
+    """An in-memory inter process message queue."""
+
+    def __init__(self, name: str, max_messages: int):
+        self.queue: q.Queue = q.Queue(max_messages)
+        self.max_messages = max_messages
+        self.name = name
+
+    def get(self, timeout: Optional[float] = None) -> bytes:
+        try:
+            message = self.queue.get(timeout=timeout)
+            self.queue.task_done()
+            return message
+        except q.Empty:
+            raise TimedOutError
+
+    def put(self, message: bytes, timeout: Optional[float] = None) -> None:
+        try:
+            return self.queue.put(message, timeout=timeout)
+        except q.Full:
+            raise TimedOutError
+
+    def unlink(self) -> None:
```

Comment:
Is it necessary that the in-memory queue is a drop-in replacement for the POSIX queue, if not this probably should raise an error, as this action is invalid for this type of queue.

Reply:
`unlink` seems like it only applies to a POSIX queue, therefore I would remove it from the abstract class and just have it in the POSIX queue class.